### ISSUE TITLE
Several `tf.math` APIs lack input checking for `bool` arguments

### ISSUE THREAD

**Issue by ArrowIntoTheSky** [tensorflow/tensorflow#54474]
Similar to [issue #54410](https://github.com/tensorflow/tensorflow/issues/54410), the following APIs also lack input validity checking for `bool` arguments.
```
tf.math.cumulative_logsumexp
tf.math.reduce_mean
tf.math.cumprod
```
Example code:
```
import tensorflow as tf
tf.math.cumulative_logsumexp([0.5, 1.0, 2.0, 4.0], axis=-1, exclusive=-1, reverse=0)
tf.math.cumprod([0.5, 1.0, 2.0, 4.0], axis=-1, exclusive=-1, reverse=0)
tf.math.reduce_mean(tf.random.uniform([1,2,2]), axis=[1,], keepdims=-1)
# Pass without exceptions
```

**Comment by Cheril311** [comment]
@ArrowIntoTheSky can you please check if there are any more arguments without validations? Will make a PR for the above issue by then

**Comment by sushreebarsa** [comment]
@ArrowIntoTheSky Could you please update as per the comment above ?Thanks!

**Comment by yongtang** [comment]
I think the issue has been fixed in https://github.com/tensorflow/tensorflow/pull/54441

**Comment by ArrowIntoTheSky** [comment]
> I think the issue has been fixed in #54441

Thanks for the PR! However, I checked on tf-nightly, and the following code still passed:
```
tf.math.reduce_mean(tf.random.uniform([1,2,2]), axis=[1,], keepdims=-1)
```
here `keepdims` should also be a `bool`, but no exception is raised.

**Comment by mohantym** [comment]
Hi @ArrowIntoTheSky ! This issue is not replicating in 2.9 version. Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/eaeef8689f15e47e5a5fcd5af36433cf/untitled370.ipynb#scrollTo=T2-hlunyu4qa) for reference. Thank you!
